File: Leetcode-Submissions/0191-number-of-1-bits/solution.cpp
class Solution {
public:
    int hammingWeight(uint32_t n) {
        int i;
          for(i =0;n != 0;n = n >> 1)
        {
        // checking last bit
        if (n & 1)
        {
            i++;
        }
    }
        return i;
    }
};

File: Leetcode-Submissions/0191-number-of-1-bits/solution.java
class Solution {
    public int hammingWeight(int n) {
        //Brute Force Approach
        // int count= 0;
        // int temp = n;
        // while(temp > 0){
        //     count += temp & 1;
        //     temp >>= 1;
        // }
        // return count;

        //brian kernighan’s algorithm
        int temp = n, count = 0;
        while(temp > 0){
            temp = temp & (temp - 1);
            count++;
        }
        return count;

    }
}
